src/Filesystem/DTO/Metadata.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace ILIAS\Filesystem\DTO;
+
+use ILIAS\Filesystem\Exception\IllegalArgumentException;
+
+/**
+ * Class Metadata
+ *
+ * This class holds all default metadata send by the filesystem adapters.
+ * Metadata instances are immutable.
+ *
+ * @author  Nicolas Schäfli <ns@studer-raimann.ch>
+ * @since 5.3
+ * @version 1.0
+ */
+final class Metadata {
+
+	/**
+	 * @var string $basename
+	 */
+	private $basename;
+	/**
+	 * @var string $path
+	 */
+	private $path;
+	/**
+	 * @var string $type
+	 */
+	private $type;
+
+
+	/**
+	 * Metadata constructor.
+	 *
+	 * @internal
+	 *
+	 * @param string $basename The basename of the file / directory.
+	 * @param string $path     The path to the file / directory.
+	 * @param string $type     The file type.
```

Comment:
The types are primarily "file" or "dir". The documentation is wrong sorry. I will fix that right away. I will add some constants for the file types.